pkg/docker/docker.go

```diff
@@ -609,15 +610,30 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	containerName := containerNameOrID(container)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+	removalDone := make(chan struct{}, 1)
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := d.RemoveContainer(container.ID); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
+		removalDone <- struct{}{}
+	}
+	go func() {
+		sigchan := make(chan os.Signal, 1)
+		signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
```

Comment:
Yes I mirrored the k8s code, basically.  monkey-see-monkey-do.  I defer to @smarterclayton for why k8s picked that set of signals and did not include SIGHUP.

Removing the stack dump is an unfortunate side effect, i can add logic back in that performs the stack dump before invoking os.Exit, but we end up in a bit of a race....one the container is removed, the logic that was waiting on the container proceeds and actually ends up exiting the program before os.Exit is even called, so the same would be true for stack dumping.  So it's going to require much more sophisticated synchronization/signaling to ensure we can execute the stack dump before proceeding.